### ISSUE TITLE
Possible problematic example in the guide "Automatic Differentiation"

### ISSUE THREAD

**Issue by szcf-weiya** [tensorflow/tensorflow#45181]
## URL(s) with the issue: 

https://www.tensorflow.org/guide/autodiff

## Description of issue (what needs changing):

In the calculation of gradient by passing a dictionary of variables, 

```python
my_vars = {
    'w': tf.Variable(tf.random.normal((3, 2)), name='w'),
    'b': tf.Variable(tf.zeros(2, dtype=tf.float32), name='b')
}

grad = tape.gradient(loss, my_vars)
grad['b']
```

It seems inappropriate since `my_vars` creates two new variables, but the pre-defined `loss` is not calculated from these two variables, and it turns out that no output from `grab['b']`. 

I think there might be two possible modifications,

- use the pre-define `w` and `b` in the dictionary

```python
my_vars = {
    'w': w,
    'b': b
}
grad = tape.gradient(loss, my_vars)
grad['b']
```

- define a new loss using `my_vars`

```python
with tf.GradientTape(persistent=True) as tape:
    y = x @ my_vars['w'] + my_vars['b']
    loss = tf.reduce_mean(y**2)
grad = tape.gradient(loss, my_vars)
grad['b']
```

Note that `persistent=True` is particularly specified, so I guess the first choice might more agree with the author's initial purpose.

**Comment by amahendrakar** [comment]
@szcf-weiya,
Can we close this issue as the linked PR is merged? Thanks!

**Comment by szcf-weiya** [comment]
@amahendrakar OK, thanks!
